### PR TITLE
Bump minimum supported Rust version

### PR DESCRIPTION
**Pull request by Minoru**
All in accordance with the schedule agreed upon in #709. I also took an opportunity to bump the GCC against which MSRV is tested.

Reviews are welcome! Will merge in three days if no issues are raised.

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -118,11 +118,11 @@ macos_task:
               RUST: stable
         # This job tests our minimum supported Rust version, so only bump on
         # Newsboat release day
-        - name: macOS GCC Rust 1.40.0
+        - name: macOS GCC Rust 1.42.0
```
Comment (dennisschagt) [inline .cirrus.yml:121]: I got a bit confused by the commit message stating that this bumps the MSRV to 1.40.0 (instead of 1.42.0 which is shown here)

Comment (Minoru) [reply .cirrus.yml:121]: Oh, looks like I wasn't paying enough attention. Thanks! I'll have to force-push to amend that.

**Comment by Minoru** [comment]
Force-pushed a reworded commit.

**Comment by Minoru** [comment]
Actually, I think I should've merged this before adding Dependabot.

Thanks for the review, @dennisschagt!

**Pull request merged**